helix-core/src/syntax.rs

```diff
@@ -126,6 +129,22 @@ pub struct LanguageServerConfiguration {
     pub language_id: Option<String>,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct FormatterConfiguration {
+    pub command: String,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub args: Vec<String>,
+    pub r#type: FormatterType,
```

Comment:
You can also name the field something like `type_` and then use [serde's rename feature](https://serde.rs/field-attrs.html#rename):

```suggestion
	#[serde(rename = "type")]
    pub type_: FormatterType,
```